aeron-system-tests/src/test/java/io/aeron/SessionSpecificPublicationTest.java

```diff
@@ -155,4 +158,53 @@ public void shouldNotCreateNonExclusivePublicationsWithDifferentSessionIdsForThe
             }
         });
     }
+
+    @Test
+    @Timeout(10)
+    @SlowTest
+    void shouldNotAddPublicationWithSameSessionUntilLingerCompletes()
+    {
+        final String channel = new ChannelUriStringBuilder()
+            .media(UDP_MEDIA)
+            .endpoint(ENDPOINT)
+            .sessionId(SESSION_ID_1)
+            .build();
+
+        final Publication publication1 = aeron.addPublication(channel, STREAM_ID);
+        final int positionLimitId = publication1.positionLimitId();
+        assertEquals(CountersReader.RECORD_ALLOCATED, aeron.countersReader().getCounterState(positionLimitId));
+
+        publication1.close();
+
+        assertThrows(RegistrationException.class, () ->
```

Comment:
This is a race so likely to be non-deterministic. Better to create a subscription and wait for it to connect but don't drain it.

Reply:
To clarify that would be a spy subscription and spies simulate connection enabled on the driver with one offer. The publication would then go into draining after close.